docs/tools/builder/index.rst

```diff
@@ -0,0 +1,329 @@
+                "code": <account code>,
+                "storage": {
+                    <storage slot>: <storage value>
+                }
+            }
+        }
+
+    To avoid unnecessary nesting especially if only few fields per account are specified, the following
+    and similar formats are possible as well:
+
+    .. code-block:: python
+
+        (address, "balance", <account balance>)
+        (address, "storage", <storage slot>, <storage value>)
+        (address, "storage", {<storage slot>: <storage value>})
+        (address, {"balance", <account balance>})
+
+
+.. autofunction:: eth.tools.builder.state.execution
+
+    For VM tests, this function specifies the code that is being run as well as the current state of
+    the EVM. State tests don't support this object. The parameter is a dictionary specifying some or
+    all of the following keys:
+
+    +--------------------+------------------------------------------------------------+
+    |  key               | description                                                |
+    +====================+============================================================+
+    | ``"address"``      | the address of the account executing the code              |
+    +--------------------+------------------------------------------------------------+
+    | ``"caller"``       | the caller address                                         |
+    +--------------------+------------------------------------------------------------+
+    | ``"origin"``       | the origin address (defaulting to the caller address)      |
+    +--------------------+------------------------------------------------------------+
+    | ``"value"``        | the value of the call                                      |
+    +--------------------+------------------------------------------------------------+
+    | ``"data"``         | the data passed with the call                              |
+    +--------------------+------------------------------------------------------------+
+    | ``"gasPrice"``     | the gas price of the call                                  |
+    +--------------------+------------------------------------------------------------+
+    | ``"gas"``          | the amount of gas allocated for the call                   |
+    +--------------------+------------------------------------------------------------+
+    | ``"code"``         | the bytecode to execute                                    |
+    +--------------------+------------------------------------------------------------+
+    | ``"vyperLLLCode"`` | the code in Vyper LLL (compiled to bytecode automatically) |
+    +--------------------+------------------------------------------------------------+
+
+
+.. autofunction:: eth.tools.builder.state.expect
+
+    This specifies the expected result of the test.
+
+    For state tests, multiple expectations can be given, differing in the the transaction data, gas
```

Comment:
> multiple expectations

Worth clarifying that these will be run as separate tests (not sequentially, right?).

Reply:
Honestly not sure, this is me porting the README which is a bit out-of-date.  I think I'm going to separate the two since I've realized they do fundamentally different things.

- one builds fixtures for VM and State tests
- one builds chains
- <not yet build thing> builds fixtures for Blockchain tests (on top of the chain builder APIs).

Reply:
>differing in ~~the~~ the transaction data

typo